### ISSUE TITLE
Why is EncryptAlgorithm version iteration so strange?😭😭😭

### ISSUE THREAD

**Issue by AntonyCheng** [apache/shardingsphere#29391]
## Question

In user manual versions 5.3.2, 5.4.0 and 5.4.1, the data encryption function is constantly changing. The MD5 encryption algorithm in version 5.3.2 was kicked out of the standard encryption method in version 5.4.X and became Assistant encryption method. In version 5.4.1, the CHAR_DIGEST_LIKE algorithm in version 5.3.X was removed. Why is there such a version iteration?

**Comment by strongduanmu** [comment]
Hi @AntonyCheng, these changes are to adapt to the rapid development of ShardingSphere. The MD5 algorithm is irreversible and functionally unsuitable as a standard encryption algorithm. Besides, since CHAR_DIGEST_LIKE is not a universal implementation algorithm, we moved it to the ShardingSphere Plugin repository after discussion on the dev mailing list to avoid the ShardingSphere main repository from being too bloated.

**Comment by AntonyCheng** [comment]
Thanks @[strongduanmu](https://github.com/strongduanmu),I have always thought this framework is very cool. It can provide convenient read-write separation and sub-library and sub-table. Some of my recent projects use version 5.3.2 as the JDBC engine. Recently, some projects need to be upgraded to Spring Boot 3, so Various components need to be updated. If it is what you mean, do I only need to import these ShardingSphere Plugin dependencies to use all the functions of version 5.4.1?

**Comment by strongduanmu** [comment]
@AntonyCheng Yes, you can copy any algorithm that you need in ShardingSphere plugin into your project. They can work as well as the previous version.

**Comment by AntonyCheng** [comment]
I copied the relevant code into my own project, but it still reported an error: 'SPI-00001: No implementation class load from SPI `org.apache.shardingsphere.encrypt.spi.EncryptAlgorithm` with type `CHAR_DIGEST_LIKE`.'
``` java
@EqualsAndHashCode
public final class CharDigestLikeEncryptAlgorithm implements LikeEncryptAlgorithm {

    private static final String DELTA_KEY = "delta";

    private static final String MASK_KEY = "mask";

    ......
}
```
How do I copy it specifically?

**Comment by strongduanmu** [comment]
You shoud config spi in META-INF/services conf, you can get spi config file in ShardingSphere plugin repository.

**Comment by AntonyCheng** [comment]
Thanks！！😊😊😊
